### PR TITLE
feature(unregister): create a function that can unregister 

### PR DESCRIPTION
**Pull request by jhionan**
Create a way to remove registered instances, also provide a way to dispose before unregister.
Create tests for this new feature.

Signed-off-by: Jhionan <jhionan@gmail.com>

### CODE REVIEW

**Comment by jhionan** [comment]
closes #3 
not only unregister singleton but any instance registered

**Comment by escamoteur** [comment]
Cool, I'll have a look at it tomorrow.
Just curious what's your use case for it?

**Comment by jhionan** [comment]
To inject some blocs like sign up and login, my user only use they once at
app, this will dispose the resources of this blocs and also remove this
instance from memory. Of course this only happens first time user open my
app, otherwise lazy initiation will not even construct.

**Comment by jhionan** [comment]
if this PR proceeds, need to update docs and readme

File: lib/get_it.dart [hunk 1]
```diff
@@ -136,6 +136,20 @@ class GetIt {
...
+  void unregister<T>({String instanceName, Function(T) disposal}) {
+    assert(_factoriesByName.containsKey(instanceName) || _factories.containsKey(T), 'Nor Type ${T.toString()} or instance Name must not be null');
+    if (instanceName == null) {
+      disposal(get<T>());
+      _factories.remove(T);
+    } else {
+      disposal(get(instanceName));
+      _factoriesByName.remove(instanceName);
+    }
+    _factories.remove(T);
```
Comment (escamoteur) [inline lib/get_it.dart]: Isn't this one remove to much?

File: lib/get_it.dart [hunk 2]
```diff
@@ -136,6 +136,20 @@ class GetIt {
       _factoriesByName[instanceName] = serviceFactory;
     }
   }
+/// Unregister by Type [T] or by name [instanceName]
+/// if you need to dispose any resources you can do it using [disposal] function
+/// that provides a instance of your class to be disposed
+  void unregister<T>({String instanceName, Function(T) disposal}) {
```
Comment (escamoteur) [inline lib/get_it.dart]: `disposal`should be optional. Consider naming it `disposingCallback` or  `disposingFunction` you don't have to make it named parameter just an optional one.

Comment (jhionan) [reply lib/get_it.dart]: I think named parameters seems a little cleaner. But ok I`ll change to required / optional parameters, and renaming disposal to disposingFunction

File: lib/get_it.dart (hunk 1 above)
Comment (jhionan) [reply lib/get_it.dart]: Yeah! Absolutely,  I missed this "_factories.remove(T);" after refactoring. my bad.

**Pull request closed without merging**